### ISSUE TITLE
"There was an error turning the model into an object" message too cryptic

### ISSUE THREAD

**Issue by fabiosantoscode** [balderdashy/waterline#1289]
When a random property is attached to the model, and this property can't be `JSON.stringify`ed, the error message `Converting circular structure to JSON` is replaced by the above message, which is a lot less descriptive of what happened.

I propose that this message is prepended to the real message instead of being replaced.

Thank you! :D

**Comment by particlebanana** [comment]
Thanks @fabiosantoscode could you post an example so I can recreate it?

**Comment by Ericat** [comment]
Example:

```
   someInstanceMethod: function() {
      this._fsm = new StateMachine({
        model: this,
        initialState: this.status
      });
    }
```

had to become the following, in order to work:

    someInstanceMethod: function() {
      var machine = new StateMachine({
        model: this,
        initialState: this.status
      });

      Object.defineProperty(this, '_fsm', {
        value: machine,
        enumerable: false,
        writeable: true
      });
    }

is that ok? Thanks!

**Comment by sailsbot** [comment]
Thanks for posting, @fabiosantoscode.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
